src/Core/ViewHelper/ViewHelperVariableContainer.php

```diff
@@ -29,6 +30,53 @@ class ViewHelperVariableContainer
      */
     protected $view;
 
+    /**
+     * Push a new delegate variable container to a stack. Done before you start
+     * sub-rendering with a ViewHelper that requires a special container for
+     * variables which is capable of being used in multidepth stacks without
+     * overriding the last instance or last set of variables.
+     *
+     * @param string $viewHelperClassName
+     * @param VariableProviderInterface $variableProvider
+     */
+    public function pushDelegateVariableContainer($viewHelperClassName, VariableProviderInterface $variableProvider)
+    {
+        if (!isset($this->objects[$viewHelperClassName]['delegateVariableProviderStack'])) {
+            $this->objects[$viewHelperClassName]['delegateVariableProviderStack'] = [];
+        }
+        $this->objects[$viewHelperClassName]['delegateVariableProviderStack'][] = $variableProvider;
+    }
+
+    /**
+     * Get the topmost delegate variable container that was previously pushed
+     * onto the stack by pushDelegateVariableContainer().
+     *
+     * @param string $viewHelperClassName
+     * @return VariableProviderInterface|null
+     */
+    public function getTopmostDelegateVariableContainer($viewHelperClassName)
```

Comment:
I think normally the term `peek` would be used for stacks.

Reply:
Little bit hard to name this because neither "end" nor "peek" really seemed to make sense. I opted instead for a consistent naming with other "get" methods but I see a problem now: proper naming would be `getTopmostDelegateVariableProvider`, not `-Container`!